modules/light-clients/06-solomachine/light_client_module_test.go

```diff
@@ -96,110 +203,1501 @@ func (suite *SoloMachineTestSuite) TestInitialize() {
 	}
 }
 
-func (suite *SoloMachineTestSuite) TestRecoverClient() {
+func (suite *SoloMachineTestSuite) TestVerifyMembership() {
 	var (
-		subjectClientID, substituteClientID       string
-		subjectClientState, substituteClientState *solomachine.ClientState
+		clientState *solomachine.ClientState
+		path        exported.Path
+		proof       []byte
+		testingPath *ibctesting.Path
+		signBytes   solomachine.SignBytes
+		err         error
+		clientID    string
 	)
 
-	testCases := []struct {
-		name     string
-		malleate func()
-		expErr   error
-	}{
-		{
-			"success",
-			func() {
+	// test singlesig and multisig public keys
+	for _, sm := range []*ibctesting.Solomachine{suite.solomachine, suite.solomachineMulti} {
+
+		testCases := []struct {
+			name     string
+			malleate func()
+			expErr   error
+		}{
+			{
+				"success",
+				func() {},
+				nil,
 			},
-			nil,
-		},
-		{
-			"cannot parse malformed substitute client ID",
-			func() {
-				substituteClientID = ibctesting.InvalidID
+			{
+				"success: client state verification",
+				func() {
+					clientState = sm.ClientState()
+					clientStateBz, err := suite.chainA.Codec.Marshal(clientState)
```

Comment:
I think I wrote a lot of these tests in the solomachine refactor from some time ago, v7 maybe 🤔 

it makes no difference here really as we're working with test code and technically it makes no difference as the solomachine itself can store however it wants, but might be nice to change this so that it doesn't cause any confusion. (this became very apparent lately working on 02-client router changes) - we always actually store the clientstate bytes as a marshalled Any in ibc-go at least, i.e. changing this to `MarshalInterface` would conform with what `setClientState()` does within the solomachine module.

would be in favour of changing for this testcase and the consensus state one below